gcp/website/frontend_emulator.py

```diff
@@ -0,0 +1,358 @@
+      public=True,
+      import_last_modified=datetime.datetime(2024, 1, 9),
+  ).put()
+
+  osv.Bug(
+      id='SUSE-SU-2023:3313-1',
+      db_id='SUSE-SU-2023:3313-1',
+      status=1,
+      upstream_raw=[
+          "CVE-2022-40982", "CVE-2023-0459", "CVE-2023-20569", "CVE-2023-21400",
+          "CVE-2023-2156", "CVE-2023-2166", "CVE-2023-31083", "CVE-2023-3268",
+          "CVE-2023-3567", "CVE-2023-3609", "CVE-2023-3611", "CVE-2023-3776",
+          "CVE-2023-4004"
+      ],
+      source='test',
+      public=True,
+      import_last_modified=datetime.datetime(2023, 8, 14),
+      timestamp=datetime.datetime(2023, 8, 14),
+  ).put()
+
+  osv.UpstreamGroup(
+      id='SUSE-SU-2023:3313-1',
+      db_id='SUSE-SU-2023:3313-1',
+      upstream_ids=[
+          "CVE-2022-40982", "CVE-2023-0459", "CVE-2023-20569", "CVE-2023-21400",
+          "CVE-2023-2156", "CVE-2023-2166", "CVE-2023-31083", "CVE-2023-3268",
+          "CVE-2023-3567", "CVE-2023-3609", "CVE-2023-3611", "CVE-2023-3776",
+          "CVE-2023-4004"
+      ],
+      last_modified=datetime.datetime(2023, 8, 14)).put()
+
+  osv.Bug(
+      id="CVE-2021-44228",
+      db_id="CVE-2021-44228",
+      public=True,
+      last_modified=datetime.datetime(2025, 2, 4),
+      source="test",
+      timestamp=datetime.datetime(2023, 8, 14),
+  ).put()
+
+  osv.Bug(
+      id="CYCLE-ROOT-1",
+      db_id="CYCLE-ROOT-1",
+      upstream_raw=['CYCLE-ROOT-2'],
+      public=True,
+      last_modified=datetime.datetime(2025, 2, 4),
+      source="test",
+      timestamp=datetime.datetime(2023, 8, 14)).put()
+  osv.Bug(
+      id="CYCLE-ROOT-2",
+      db_id="CYCLE-ROOT-2",
+      upstream_raw=['CYCLE-ROOT-1'],
+      public=True,
+      last_modified=datetime.datetime(2025, 2, 4),
+      source="test",
+      timestamp=datetime.datetime(2023, 8, 14)).put()
+  osv.Bug(
+      id="CYCLE-ROOT-3",
+      db_id="CYCLE-ROOT-3",
+      upstream_raw=['CYCLE-ROOT-1', 'CYCLE-ROOT-2'],
+      public=True,
+      last_modified=datetime.datetime(2025, 2, 4),
+      source="test",
+      timestamp=datetime.datetime(2023, 8, 14)).put()
+
+  osv.UpstreamGroup(
+      id='CYCLE-ROOT-3',
+      db_id='CYCLE-ROOT-3',
+      upstream_ids=['CYCLE-ROOT-1', 'CYCLE-ROOT-2'],
+      last_modified=datetime.datetime(2023, 8, 14)).put()
+  osv.UpstreamGroup(
+      id='CYCLE-ROOT-1',
+      db_id='CYCLE-ROOT-1',
+      upstream_ids=['CYCLE-ROOT-2'],
+      last_modified=datetime.datetime(2023, 8, 14)).put()
+
+  osv.UpstreamGroup(
+      id='CYCLE-ROOT-2',
+      db_id='CYCLE-ROOT-2',
+      upstream_ids=['CYCLE-ROOT-1'],
+      last_modified=datetime.datetime(2023, 8, 14)).put()
+
+
+if __name__ == '__main__':
+  ds_emulator = tests.start_datastore_emulator()
+  import main
+  try:
+    with ndb.Client().context() as context:
+      context.set_memcache_policy(False)
+      context.set_cache_policy(False)
+      setUp()
+      main.app.run(host='127.0.0.1', port=8000, debug=False)
+    # main.app.run()
```

Comment:
nit: remove commented line
I also think `app.run` can go outside the `with ... as context`